service.php

```diff
@@ -263,6 +259,20 @@ function get_credentials() {
 		return apply_filters( 'keyring_service_credentials', $creds, $this->get_name() );
 	}
 
+		/**
+	 * Set the basic credential properties based
+	 * on an Array of credentials
+	 *
+	 * @param Mixed $credentials
+	 */
+	function init_credentials( $credentials ) {
+		if ( ! empty( $credentials ) && is_array( $credentials ) ) {
```

Comment:
Could we make the `$credentials` parameter optional and if it's empty, call `get_credentials`? It might make the service classes even tidier.

Reply:
Good call, this makes things much cleaner (81ae051b0ff362ba9ccc2e7c0cc33377b4592964)